libsolutil/JSON.h

```diff
@@ -67,4 +67,67 @@ std::string jsonPrint(Json::Value const& _input, JsonFormat const& _format);
 /// \return \c true if the document was successfully parsed, \c false if an error occurred.
 bool jsonParseStrict(std::string const& _input, Json::Value& _json, std::string* _errs = nullptr);
 
+namespace detail
+{
+
+template<typename T>
+struct helper;
+
+#define DEFINE_HELPER(TYPE, CHECK_TYPE, CONVERT_TYPE)                                                     \
+	template<>                                                                                            \
+	struct helper<TYPE>                                                                                   \
+	{                                                                                                     \
+		static bool ofType(Json::Value const& _input, std::string const& _name)                           \
+		{                                                                                                 \
+			return _input[_name].CHECK_TYPE();                                                            \
+		}                                                                                                 \
+		static TYPE get(Json::Value const& _input, std::string const& _name)                              \
+		{                                                                                                 \
+			return _input[_name].CONVERT_TYPE();                                                          \
+		}                                                                                                 \
+		static TYPE getOrDefault(Json::Value const& _input, std::string const& _name, TYPE _default = {}) \
+		{                                                                                                 \
+			TYPE result = _default;                                                                       \
+			if (helper::ofType(_input, _name))                                                            \
+				result = _input[_name].CONVERT_TYPE();                                                    \
+			return result;                                                                                \
+		}                                                                                                 \
+	};
+
+DEFINE_HELPER(float, isDouble, asFloat)
+DEFINE_HELPER(double, isDouble, asDouble)
+DEFINE_HELPER(std::string, isString, asString)
+DEFINE_HELPER(Json::Int, isInt, asInt)
+DEFINE_HELPER(Json::Int64, isInt64, asInt64)
+DEFINE_HELPER(Json::UInt, isUInt, asUInt)
+DEFINE_HELPER(Json::UInt64, isUInt64, asUInt64)
+
+} // namespace detail
+
+template<typename T>
+bool ofType(Json::Value const& _input, std::string const& _name)
+{
+	return detail::helper<T>::ofType(_input, _name);
 }
+
+template<typename T>
+bool ofTypeIfExists(Json::Value const& _input, std::string const& _name)
+{
+	if (_input.isMember(_name))
+		return ofType<T>(_input, _name);
+	return true;
+}
+
+template<typename T>
+T get(Json::Value const& _input, std::string const& _name)
```

Comment:
Won't it interfere with `std::get()` used e.g. for variants? We do have `using namespace std` in a lot of places.

I think it might be better to put these helpers into a namespace. This would also make them easier to understand. Their names are pretty generic but they're only usable with JSON, so I think something like `Json::get` would be more appropriate and unambiguous.

Reply:
I think there should be no conflict possible. `std::get` seem to operate on `std::array`, `std::pair`, `std::variant` and `std::ranges::subrange`, where these types are always used as the first parameters.

Reply:
Maybe. Overloading rules for templates have some weird exceptions about what can actually overload and what shadows other things. That's what I'm mostly worried about.

Reply:
In any case, I think the name is just to generic. I'd namespace it even just for that.

@nikola-matic What's your opinion here?

Reply:
I don't think it would be a problem in this case as ADL will sort this out anyway, but I do think it would be nice to shove this into a namespace. @ekpyron, thoughts?

Reply:
It is already in ``solidity::util`` (which I'd tend not to fully import anyways - I know that it's done almost 200 times in the code, though ;-))... and it has concrete types, so it'll always be the most specialized overload... and I don't see any want for calling ``std::get`` on a ``Json::Value`` - and even then it'd be possible by qualifying.

So in the end: it's fine.

Reply:
Not that the name is perfect - but I don't want to stall this for a few round trips due to naming. It won't cause issues as is and we can still rename after all this stuff is merged, if we're bothered by it.